### PR TITLE
Hotfix/problem submission form

### PR DESCRIPTION
**Pull request by thocle86**
* Addition of asserts in the home and keysVision entities
* Addition of HiddenType in the formType to avoid fatal errors => to add asserts on entities that do not have certain properties
* Addition of try-catch blocks to avoid fatal errors when not adding an image => a simple required on the input image obliges to give an image again during the modification

### CODE REVIEW

**Review by BoulMaker — DISMISSED** [review]

File: src/Controller/PurposeValuesController.php [hunk 1]
```diff
@@ -29,32 +30,33 @@ public function newPurposeValues(
         string $type
     ): Response {
         $home = new Home();
+        $home->setType($type);
+        $home->setTitle("null");//because there is no title
+        $home->setPictureTwo("null");//because there is no picture2
+        $home->setPictureThree("null");//because there is no picture3
+        $home->setLegendPictureTwo("null");//because there is no picture2
+        $home->setLegendPictureThree("null");//because there is no picture3
```
Comment (Quentin-BLANC) [inline src/Controller/PurposeValuesController.php:38]: because there is no LEGEND picture ?

**Review by Quentin-BLANC — DISMISSED** [review]

File: src/Controller/PurposeValuesController.php (hunk 1 above)
Comment (ghost) [reply src/Controller/PurposeValuesController.php:38]: There is only one picture in purpose article and values article so there is lo legend

File: src/Controller/CarouselController.php [hunk 2]
```diff
@@ -28,39 +29,52 @@ public function newCarousel(
...
+        $carousel->setText('null');//because there is no text in a carousel
+        $form = $this->createForm(CarouselType::class, $carousel);
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            $pictureOne = $form->get('pictureOne')->getData();
+            $pictureTwo = $form->get('pictureTwo')->getData();
+            $pictureThree = $form->get('pictureThree')->getData();
+            try {
+                if (!isset($pictureOne)) {
+                    throw new Exception('⚠️ ATTENTION: vous devez ajouter 3 photos pour valider le formulaire');
```
Comment (RomainClair) [inline src/Controller/CarouselController.php]: Exception are probably not good way to warn the application user for errors

**Review by RomainClair — CHANGES_REQUESTED** [review]

**Review by CeliaBoiss — APPROVED** [review]

File: src/Entity/KeysVision.php [hunk 3]
```diff
@@ -82,58 +79,63 @@ class KeysVision
...
+     *      maxMessage = "Votre texte ne peut pas faire plus de {{ limit }} caractères"
      * )
      */
     private string $text4;
 
     /**
      * @ORM\Column(type="string", length=50, nullable=true)
+     * @Assert\NotBlank(message="Ce champ ne peut pas être vide")
      * @Assert\Length(
-     *      min = 2,
```
Comment (Quentin-BLANC) [inline src/Entity/KeysVision.php:94]: Why here is min=1 and max=5, and before is min=2 and max=50.
And it's write length=50.

**Review by BoulMaker — APPROVED** [review]

**Review by Quentin-BLANC — APPROVED** [review]

**Review by RomainClair — APPROVED** [review]

**Pull request merged**